src/gm-apclientpp.cpp

```diff
@@ -222,34 +235,49 @@ double apclient_connect(const char* uuid, const char* game, const char* host)
             std::string game = apclient->get_player_game(apclient->get_player_number());
             std::string s;
             int index = items.front().index;
+            auto j = json{
+                { "index", index }
+            };
             int i = 0;
             for (const auto& item: items) {
-                std::string item_name = apclient->get_item_name(item.item, game);
+                std::string item_name = escape_string(apclient->get_item_name(item.item, game));
                 s +=
                     "global.arg_ids[" + std::to_string(i) + "]=" + std::to_string(item.item) + ";\r\n"
-                    "global.arg_names[" + std::to_string(i) + "]='" + escape_string(item_name) + "';\r\n"
+                    "global.arg_names[" + std::to_string(i) + "]='" + item_name + "';\r\n"
                     "global.arg_flags[" + std::to_string(i) + "]=" + std::to_string(item.flags) + ";\r\n"
                     "global.arg_players[" + std::to_string(i) + "]=" + std::to_string(item.player) + ";\r\n"
                     "global.arg_locations[" + std::to_string(i) + "]=" + std::to_string(item.location) + ";\r\n";
+                j["ids"][i] = item.item;
+                j["names"][i] = item_name;
```

Comment:
It definitely should not be escaped for the json data, since the string fetching calls already escape strings.

You are right that this was leading `\n` to become `#` and then `\#`, upon being double escaped. Similarly, it was turning `#` into `\#` and then `\\#`. It was even worse for `'`, which should not have been escaped at all for the json data.